mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -213,52 +213,82 @@ LogicalResult ConvConverter<ConvType>::matchAndRewrite(
   SmallVector<int64_t> NCHW2NHWC{0, 2, 3, 1};
   SmallVector<int64_t> NHWC2NCHW{0, 3, 1, 2};
 
+  int dims = outputTy.getShape().size() - 2;
+  SmallVector<int64_t> toChannelLast{0};
+  SmallVector<int64_t> fromChannelLast{0, dims + 1};
+  for (int i = 0; i < dims; i++) {
+    toChannelLast.push_back(i + 2);
+    fromChannelLast.push_back(i + 1);
+  }
+  toChannelLast.push_back(1);
+
   // insert transpose to input and filter tensors
-  input = getTransposeOp(loc, input, rewriter, NCHW2NHWC);
-  filter = getTransposeOp(loc, filter, rewriter, NCHW2NHWC);
+  input = getTransposeOp(loc, input, rewriter, toChannelLast);
+  filter = getTransposeOp(loc, filter, rewriter, toChannelLast);
   ArrayRef<int64_t> outShape = outputTy.getShape();
 
   // original output shape was NCHW, change it into NHWC
-  SmallVector<int64_t> newShape{outShape[0], outShape[2], outShape[3],
-                                outShape[1]};
+  SmallVector<int64_t> newShape{outShape[0]};
+  for (int i = 0; i < dims; i++)
+    newShape.push_back(outShape[i + 2]);
+  newShape.push_back(outShape[1]);
   Type newOutTy = RankedTensorType::get(newShape, outputTy.getElementType());
 
   // Construct a new Conv2DOp.
-  auto cop = rewriter.create<tosa::Conv2DOp>(
-      loc, newOutTy,
-      ValueRange{input, filter,
-                 getZeroTensor(
-                     loc, outputTy.getElementType(),
-                     filter.getType().template cast<ShapedType>().getShape()[0],
-                     rewriter)});
+  Operation *cop;
+  switch (dims) {
+  case 2:
+    cop = rewriter.create<tosa::Conv2DOp>(
+        loc, newOutTy,
+        ValueRange{
+            input, filter,
+            getZeroTensor(
+                loc, outputTy.getElementType(),
+                filter.getType().template cast<ShapedType>().getShape()[0],
+                rewriter)});
+    break;
+  case 3:
+    cop = rewriter.create<tosa::Conv3DOp>(
+        loc, newOutTy,
+        ValueRange{
+            input, filter,
+            getZeroTensor(
+                loc, outputTy.getElementType(),
+                filter.getType().template cast<ShapedType>().getShape()[0],
+                rewriter)});
+    break;
+  default:
```

Comment:
... Add 1D?

Reply:
Not sure what to rewrite it into, since there isn't a tosa::Conv1DOp.